### PR TITLE
Remove the diagnostic server from flutter_tools

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
Nice! More deleted un[der]tested code == better test coverage too :)
lgtm

**Comment by devoncarew** [comment]
@jason-simmons, DBC: you can also remove the searchStep field from PortScanner.findPreferredPort. It defaulted to 2 because we were binding two two ports (the observatory and diagnostics services).

File: packages/flutter_tools/lib/src/commands/screenshot.dart [hunk 1]
```diff
@@ -29,14 +27,7 @@ class ScreenshotCommand extends FlutterCommand {
...
       help: 'Retrieve the last frame rendered by a Flutter app as a Skia picture\n'
-        'using the specified diagnostic server port.\n'
-        'To find the diagnostic server port number, use "flutter run --verbose"\n'
-        'and look for "Diagnostic server listening on" in the output.'
-    );
-    argParser.addOption(
-      _kSkiaServe,
-      valueHelp: 'url',
-      help: 'Post the picture to a skiaserve debugger at this URL.',
+        'using the specified observatory port.\n'
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/screenshot.dart]: we should probably keep the help information about how to obtain the observatory port

Comment (Hixie) [reply packages/flutter_tools/lib/src/commands/screenshot.dart]: (suitably updated of course)

Comment (jason-simmons) [reply packages/flutter_tools/lib/src/commands/screenshot.dart]: done

**Pull request merged**